### PR TITLE
Adds the `report` subcommand for all plugin

### PR DESCRIPTION
**Pull request by mbtamuli**
- [x] apps:report
- [ ] checks:report
- [ ] docker-options:report
- [ ] domains:report
- [ ] nginx:report
- [ ] proxy:report
- [x] ps:report
- [ ] storage:report

This is one PR for #2356

### CODE REVIEW

**Comment by mbtamuli** [comment]
I realized later after pushing this, that this PR is incomplete. I have to edit the `apps/command` file and also edit the docs.

File: plugins/apps/subcommands/report [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  if [[ -z "$INFO_FLAG" ]]; then
+    dokku_log_info2 "$APP"
+    if (is_deployed "$APP"); then
+      for flag in "${flag_map[@]}"; do
+        key="$(echo "${flag#--}" | cut -f1 -d' ' | tr - ' ')"
+        dokku_log_verbose "$(printf "%-20s %-25s" "${key^}" "${flag#*: }")"
+      done
+    else
+      echo "not deployed"
```
Comment (josegonzalez) [inline plugins/apps/subcommands/report]: Should use our error helpers. If there isnt one that doesnt output to stderr without exiting, create one :)

Comment (mbtamuli) [inline plugins/apps/subcommands/report]: Done. Should I squash the commits?

**Comment by mbtamuli** [comment]
@josegonzalez Can you also guide me to write tests for these?

**Comment by josegonzalez** [comment]
You can just check the specific output of commands, and also that the flags all work properly. Here is an [example](https://github.com/dokku/dokku-redis/blob/master/tests/service_info.bats#L36).

**Comment by josegonzalez** [comment]
@mbtamuli Which one do you want to tackle next?

**Comment by mbtamuli** [comment]
@josegonzalez I don't know if there's any order to it. Maybe `proxy:report` if there's no specific order?

About `dokku report APP` or `dokku report`, to get a complete report of everything, it will have to be done after completing the report subcommand for all the plugins mentioned.

**Comment by josegonzalez** [comment]
@mbtamuli follow on with `proxy:report` :)

**Comment by josegonzalez** [comment]
For the `dokku report APP` etc., yes I agree. Once the other plugins have a `report` subcommand, will be fairly trivial to do this those.

**Comment by mbtamuli** [comment]
I will try to follow up on `proxy:report` soon.

**Pull request merged**